cpp/tests/utilities/thrust_wrapper.cu

```diff
@@ -374,6 +375,151 @@ template rmm::device_uvector<int32_t> unique<int32_t>(raft::handle_t const& hand
 template rmm::device_uvector<int64_t> unique<int64_t>(raft::handle_t const& handle,
                                                       rmm::device_uvector<int64_t>&& values);
 
+template <typename value_t>
+size_t unique_count(raft::handle_t const& handle,
+                    cugraph::dataframe_buffer_type_t<value_t> const& values)
+{
+  return thrust::unique_count(handle.get_thrust_policy(),
+                              cugraph::get_dataframe_buffer_begin(values),
+                              cugraph::get_dataframe_buffer_end(values));
+}
+
+template size_t unique_count<int32_t>(raft::handle_t const& handle,
+                                      rmm::device_uvector<int32_t> const& values);
+
+template size_t unique_count<int64_t>(raft::handle_t const& handle,
+                                      rmm::device_uvector<int64_t> const& values);
+
+template <typename key_t, typename value_t>
+std::tuple<cugraph::dataframe_buffer_type_t<key_t>, cugraph::dataframe_buffer_type_t<value_t>>
+reduce_by_key(raft::handle_t const& handle,
+              cugraph::dataframe_buffer_type_t<key_t>&& keys,
+              cugraph::dataframe_buffer_type_t<value_t>&& values,
```

Comment:
Not sure these should be R-values. thrust::reduce_by_key does not destroy the inputs, I think this can just be const& similar to unique_count.

Reply:
Right `thrust::reduce_by_key` stores the result in different arrays